google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Range.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import java.io.Serializable;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class Range implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  /** Returns the start value of the range. A null value represents an unbounded start. */
+  @Nullable
+  public abstract String getStart();
```

Comment:
A usability question:  currently the Range object returns the string representation, which feels limiting.  Can we do something like have the getters return FieldValue?  That would make it possible for users to do things like range.getStart().getTimestampInstant() for example with a `RANGE<TIMESTAMP>`.

Reply:
Updated the getters to return FieldValue. PTAL.